src/scripts/main.js

```diff
@@ -1 +1,108 @@
 'use strict';
+
+let isClickedLeft = false;
+let isClickedRight = false;
+let timer;
+const notification = document.createElement('div');
+
+notification.setAttribute('data-qa', 'notification');
+
+function resetTimer() {
+  clearTimeout(timer);
+
+  timer = setTimeout(() => {
+    if (!isClickedLeft) {
+      firstPromise();
+    }
+  }, 3000);
+}
+
+function handleFirstClick() {
+  firstPromise();
+  resetTimer();
+
+  document.removeEventListener('click', handleFirstClick);
+}
+
+document.addEventListener('click', handleFirstClick);
+
+document.addEventListener('mousedown', (e) => {
+  if (e.button === 0) {
+    isClickedLeft = true;
+  }
+
+  if (e.button === 2) {
+    isClickedRight = true;
+  }
+
+  secondPromise();
+
+  if (isClickedLeft && isClickedRight) {
+    thirdPromise();
+  }
+});
+
+document.addEventListener('contextmenu', (e) => {
+  e.preventDefault();
+});
+
+function firstPromise() {
+  return new Promise((resolve, reject) => {
+    if (isClickedLeft) {
+      resolve('First promise was resolved');
+    } else {
+      reject(new Error('First promise was rejected'));
+    }
+  })
+    .then((result) => {
+      if (!notification.classList.contains('success')) {
```

Comment:
The class `success` is added to the `notification` element without being defined. It is recommended to define this class in your CSS.